src/vm/analysis/errors.rs

```diff
@@ -131,6 +131,7 @@ pub enum CheckErrors {
     // expect a function, or applying a function to a list
     NonFunctionApplication,
     ExpectedListApplication,
+    ExpectedBuffer16(TypeSignature),
```

Comment:
Probably I should use an existing error like "TypeError(TypeSignature, TypeSignature)" instead of this.

Reply:
Yes -- if you use `checker.type_check_expects(..)` to perform the type checks, then it will create the `TypeError(..)` error for you, so you should be able to remove the `ExpectedBuffer16` variant.

Reply:
The error in 'src/vm/functions/conversions.rs' is now:

                return Err(CheckErrors::TypeError(
                    SequenceType(BufferType(BufferLength(16))),
                    TypeSignature::type_of(&value),
                )